R/utils.R

```diff
@@ -193,3 +193,84 @@ write_rdump <- function(data) {
   )
   temp_file
 }
+
+#' Cleanup build files of a Stan model
+#'
+#' deletese the model_name.o, model_name.hpp and the executable.
+#'
+#' @param model_path (string) The absolute path to the model
+#' @param remove_main (logical) Set TRUE to also remove the cmdstan main.o
+#' @noRd
+build_cleanup <- function(model_path,
+                          remove_main = FALSE) {
+  model_hpp_file = paste(model_path, ".hpp", sep = "")
+  model_o_file = paste(model_path, ".o", sep = "")
+  if(file.exists(model_hpp_file)) {
+    file.remove(model_hpp_file)
+  }
+  if(file.exists(model_o_file)) {
+    file.remove(model_o_file)
+  }
+  if(file.exists(model_path)) {
+    file.remove(model_path)
+  }
+  if(remove_main) {
+    main_o_file <- paste(cmdstan_path(), "src", "cmdstan", "main.o", sep = "/")
+    if(file.exists(main_o_file)) {
+      file.remove(main_o_file)
+    }
+  }
+}
+
+set_make_local <- function(threads = FALSE,
+                           opencl = FALSE,
+                           opencl_platform_id = 0,
+                           opencl_device_id = 0,
+                           compiler_flags = NULL) {
+  make_local_path <- paste(cmdstan_path(), "make", "local", sep = "/")
+  old_make_local_content <- ""
+  if(file.exists(make_local_path)) {
+    # read the contents of make/local to compare with the new contents
+    old_make_local_content <- paste(readLines(make_local_path), collapse = "\n")
+  }
+  if(opencl) {
+    stan_opencl <- "STAN_OPENCL = true"
+    platform_id <- paste("OPENCL_PLATFORM_ID", opencl_platform_id, sep = " = ")
+    device_id <- paste("OPENCL_DEVICE_ID", opencl_device_id, sep = " = ")
+    compiler_flags <- c(compiler_flags, stan_opencl, platform_id, device)
```

Comment:
Does this work for you? I think the object `device` that is included as the last element of `compiler_flags` is not defined  (just `device_id`), so I would think this errors. Or am I missing something?

Reply:
Yeah, I switched it up at the last second. Will fix.

Reply:
Fixed.